### ISSUE TITLE
Mismatch between the support library versions

### ISSUE THREAD

**Issue by Jun711** [dpa99c/cordova-diagnostic-plugin#232]
sorry for the lengthy explanation. 
I recently updated this plugin version to 3.6.7 from 3.1.0. After that, I couldn't do cordova build android. 
In the end, I found out the error arise from android support library version clashes between this plugin's and this onesignal [plugin ](https://github.com/OneSignal/OneSignal-Cordova-SDK/blob/00de4d34e1dc5c44877899150b78cc23e6a081e2/plugin.xml#L37) 

The following is the error message

       :processArmv7DebugManifest FAILED
        
        FAILURE: Build failed with an exception.
        
        * What went wrong:
        Execution failed for task ':processArmv7DebugManifest'.
        > Manifest merger failed : Attribute meta-data#android.support.VERSION@value value=(25.4.0) from 
        [com.android.support:appcompat-v7:25.4.0] AndroidManifest.xml:28:13-35
           	is also present at [com.android.support:customtabs:26.1.0] AndroidManifest.xml:25:13-35 value=(26.1.0).
           	Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to 
        override.
        
        * Try:
        Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
        
        BUILD FAILED`

I did try with this plugin  cordova-android-support-gradle-release but it gave me some other errors. 

Then, I removed one signal plugin and tried to build again, the support library version clashes among this plugin and two other plugins: [camera plugin](https://github.com/apache/cordova-plugin-camera/blob/16c4325fef80327bceccedb491e29064da79b556/plugin.xml#L94) and [social sharing ](https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin/blob/38920024ae57cd2dbbb8b1b05d61459416dd40a5/plugin.xml#L76)

The following is the error message:

         FAILURE: Build failed with an exception.
        
        * What went wrong:
        Execution failed for task ':processArmv7DebugManifest'.
        > Manifest merger failed : Attribute meta-data#android.support.VERSION@value value=(25.3.1) from 
        [com.android.support:appcompat-v7:25.3.1] AndroidManifest.xml:27:9-31
           	is also present at [com.android.support:support-v4:26.0.0-alpha1] AndroidManifest.xml:27:9-38 
        value=(26.0.0-alpha1).
           	Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:25:5-27:34 to 
        override.
        
        * Try:
        Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
        
        BUILD FAILED

However,
It works if I use version 3.1 of this plugin which has the following support library versions:
        
        <framework src="com.android.support:support-v4:23.+" />
        <framework src="com.android.support:appcompat-v7:23.+" />

compared to the current one

        <framework src="com.android.support:support-v4:25.+" />
        <framework src="com.android.support:appcompat-v7:25.+" />

I have skimmed through all these issues:
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/203
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/211
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/212
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/205
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/161
- https://github.com/dpa99c/cordova-diagnostic-plugin/issues/168

Questions
1) How to fix this so that I can use the latest version?
2) Could you explain a little on why the latest version clashes with plugins? 
<framework src="com.android.support:support-v4:24.1.1+" /> requires a version greater than 24.1.1?
<framework src="com.android.support:support-v4:+" /> requires the lastest?

**Comment by dpa99c** [comment]
> 1. How to fix this so that I can use the latest version?

TLDR; Specify `cordova-android-support-gradle-release` when installing it:

	cordova plugin add cordova-android-support-gradle-release --variable ANDROID_SUPPORT_VERSION=24.+
	
See the console output which illustrates the fix:

<details>
<summary>console output</summary>

```

$ cordova create test && cd test
Creating a new cordova project.

$ cordova platform add android
Using cordova-fetch for cordova-android@~6.2.2
Adding android project...
Creating Cordova project for the Android platform:
        Path: platforms\android
        Package: io.cordova.hellocordova
        Name: HelloCordova
        Activity: MainActivity
        Android target: android-25
Subproject Path: CordovaLib
Android project created with cordova-android@6.2.3
Discovered plugin "cordova-plugin-whitelist" in config.xml. Adding it to the project
Installing "cordova-plugin-whitelist" for android

               This plugin is only applicable for versions of cordova-android greater than 4.0. If you have a previous platform version, you do *not* need this plugin since the whitelist will be built in.

Adding cordova-plugin-whitelist to package.json
Saved plugin info for "cordova-plugin-whitelist" to config.xml
--save flag or autosave detected
Saving android@~6.2.3 into config.xml file ...

$ cordova plugin add cordova.plugins.diagnostic
Installing "cordova.plugins.diagnostic" for android
Subproject Path: CordovaLib
Adding cordova.plugins.diagnostic to package.json
Saved plugin info for "cordova.plugins.diagnostic" to config.xml

$ cordova plugin add cordova-plugin-camera
Installing "cordova-plugin-camera" for android
Installing "cordova-plugin-compat" for android
Subproject Path: CordovaLib
Adding cordova-plugin-camera to package.json
Saved plugin info for "cordova-plugin-camera" to config.xml

$ cordova plugin add cordova-plugin-x-socialsharing
Installing "cordova-plugin-x-socialsharing" for android
Installing "es6-promise-plugin" for android
Subproject Path: CordovaLib
Adding cordova-plugin-x-socialsharing to package.json
Saved plugin info for "cordova-plugin-x-socialsharing" to config.xml

$ cordova plugin add onesignal-cordova-plugin
Installing "onesignal-cordova-plugin" for android
Subproject Path: CordovaLib
Adding onesignal-cordova-plugin to package.json
Saved plugin info for "onesignal-cordova-plugin" to config.xml

$ cordova build android
ANDROID_HOME=C:\Users\Admin\AppData\Local\Android\sdk
JAVA_HOME=C:\Program Files\Java\jdk1.8.0_101
Starting a Gradle Daemon (subsequent builds will be faster)
Parallel execution with configuration on demand is an incubating feature.
:wrapper

BUILD SUCCESSFUL

Total time: 12.047 secs
Subproject Path: CordovaLib
Starting a Gradle Daemon (subsequent builds will be faster)
Parallel execution with configuration on demand is an incubating feature.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_6nhbipi8n96npr5m0yzptpel7.run(D:\Temp\test\platforms\android\build.gradle:138)
Observed package id 'build-tools;25.0.2' in inconsistent location 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\current' (Expected 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\25.0.2')
Already observed package id 'build-tools;25.0.2' in 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\25.0.2'. Skipping duplicate at 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\current'
Observed package id 'build-tools;25.0.2' in inconsistent location 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\current' (Expected 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\25.0.2')
Already observed package id 'build-tools;25.0.2' in 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\25.0.2'. Skipping duplicate at 'C:\Users\Admin\AppData\Local\Android\sdk\build-tools\current'
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
Download https://maven.google.com/com/android/support/support-v4/25.4.0/support-v4-25.4.0.pom
Download https://maven.google.com/com/android/support/appcompat-v7/25.4.0/appcompat-v7-25.4.0.pom
Download https://maven.google.com/com/android/support/support-v4/24.1.1/support-v4-24.1.1.pom
Download https://repo1.maven.org/maven2/com/onesignal/OneSignal/3.6.1/OneSignal-3.6.1.pom
Download https://maven.google.com/com/google/android/gms/play-services-gcm/11.4.0/play-services-gcm-11.4.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-location/11.4.0/play-services-location-11.4.0.pom
Download https://maven.google.com/com/android/support/support-v4/26.1.0/support-v4-26.1.0.pom
Download https://maven.google.com/com/android/support/customtabs/26.1.0/customtabs-26.1.0.pom
Download https://maven.google.com/com/android/support/support-annotations/25.4.0/support-annotations-25.4.0.pom
Download https://maven.google.com/com/android/support/support-vector-drawable/25.4.0/support-vector-drawable-25.4.0.pom
Download https://maven.google.com/com/android/support/animated-vector-drawable/25.4.0/animated-vector-drawable-25.4.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-gcm/11.2.2/play-services-gcm-11.2.2.pom
Download https://maven.google.com/com/google/android/gms/play-services-location/11.2.2/play-services-location-11.2.2.pom
Download https://maven.google.com/com/android/support/support-v4/26.0.2/support-v4-26.0.2.pom
Download https://maven.google.com/com/android/support/customtabs/26.0.2/customtabs-26.0.2.pom
Download https://maven.google.com/com/android/support/support-compat/25.4.0/support-compat-25.4.0.pom
Download https://maven.google.com/com/android/support/support-core-ui/25.4.0/support-core-ui-25.4.0.pom
Download https://maven.google.com/com/android/support/support-media-compat/26.1.0/support-media-compat-26.1.0.pom
Download https://maven.google.com/com/android/support/support-core-utils/26.1.0/support-core-utils-26.1.0.pom
Download https://maven.google.com/com/android/support/support-fragment/26.1.0/support-fragment-26.1.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-base/11.4.0/play-services-base-11.4.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-basement/11.4.0/play-services-basement-11.4.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-iid/11.4.0/play-services-iid-11.4.0.pom
Download https://maven.google.com/com/google/android/gms/play-services-tasks/11.4.0/play-services-tasks-11.4.0.pom
Download https://maven.google.com/com/android/support/support-compat/26.1.0/support-compat-26.1.0.pom
Download https://maven.google.com/android/arch/lifecycle/runtime/1.0.0/runtime-1.0.0.pom
Download https://maven.google.com/android/arch/lifecycle/common/1.0.0/common-1.0.0.pom
Download https://maven.google.com/android/arch/core/common/1.0.0/common-1.0.0.pom
Download https://maven.google.com/com/android/support/support-core-ui/26.1.0/support-core-ui-26.1.0.pom
Download https://maven.google.com/com/android/support/support-annotations/26.1.0/support-annotations-26.1.0.pom
Download https://maven.google.com/com/android/support/support-annotations/26.1.0/support-annotations-26.1.0.jar
Download https://maven.google.com/android/arch/lifecycle/common/1.0.0/common-1.0.0.jar
Download https://maven.google.com/android/arch/core/common/1.0.0/common-1.0.0.jar
Download https://maven.google.com/android/arch/lifecycle/runtime/1.0.0/runtime-1.0.0.aar
Download https://maven.google.com/com/android/support/support-compat/26.1.0/support-compat-26.1.0.aar
Download https://maven.google.com/com/android/support/support-media-compat/26.1.0/support-media-compat-26.1.0.aar
Download https://maven.google.com/com/android/support/support-core-utils/26.1.0/support-core-utils-26.1.0.aar
Download https://maven.google.com/com/android/support/support-core-ui/26.1.0/support-core-ui-26.1.0.aar
Download https://maven.google.com/com/android/support/support-fragment/26.1.0/support-fragment-26.1.0.aar
Download https://maven.google.com/com/android/support/support-v4/26.1.0/support-v4-26.1.0.aar
Download https://maven.google.com/com/android/support/support-vector-drawable/25.4.0/support-vector-drawable-25.4.0.aar
Download https://maven.google.com/com/android/support/animated-vector-drawable/25.4.0/animated-vector-drawable-25.4.0.aar
Download https://maven.google.com/com/android/support/appcompat-v7/25.4.0/appcompat-v7-25.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-basement/11.4.0/play-services-basement-11.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-tasks/11.4.0/play-services-tasks-11.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-base/11.4.0/play-services-base-11.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-iid/11.4.0/play-services-iid-11.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-gcm/11.4.0/play-services-gcm-11.4.0.aar
Download https://maven.google.com/com/google/android/gms/play-services-location/11.4.0/play-services-location-11.4.0.aar
Download https://maven.google.com/com/android/support/customtabs/26.1.0/customtabs-26.1.0.aar
Download https://repo1.maven.org/maven2/com/onesignal/OneSignal/3.6.1/OneSignal-3.6.1.aar
:preBuild
:CordovaLib:preBuild UP-TO-DATE
:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:checkDebugManifest
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:CordovaLib:prepareDebugDependencies
:CordovaLib:compileDebugAidl
:preReleaseBuild UP-TO-DATE
:generateDebugBuildConfig
:generateDebugResValues
:CordovaLib:compileDebugNdk NO-SOURCE
:CordovaLib:compileLint
:incrementalDebugJavaCompilationSafeguard
:CordovaLib:copyDebugLint NO-SOURCE
:CordovaLib:mergeDebugShaders
:compileDebugNdk NO-SOURCE
:mergeDebugShaders
:CordovaLib:compileDebugShaders
:CordovaLib:generateDebugAssets
:CordovaLib:mergeDebugAssets
:compileDebugShaders
:generateDebugAssets
:processDebugJavaRes
:CordovaLib:mergeDebugProguardFiles
:processDebugJavaRes NO-SOURCE
:CordovaLib:packageDebugRenderscript
:validateSigningDebug
:CordovaLib:packageDebugRenderscript NO-SOURCE
:CordovaLib:compileDebugRenderscript
:CordovaLib:generateDebugResValues
:CordovaLib:generateDebugResources
:CordovaLib:packageDebugResources
:CordovaLib:processDebugManifest
:CordovaLib:generateDebugBuildConfig
:CordovaLib:processDebugResources
:CordovaLib:generateDebugSources
:CordovaLib:incrementalDebugJavaCompilationSafeguard
:CordovaLib:compileDebugJavaWithJavac
:CordovaLib:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processDebugJavaRes NO-SOURCE
:CordovaLib:transformResourcesWithMergeJavaResForDebug
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForDebug
:CordovaLib:mergeDebugJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForDebug
:CordovaLib:transformNative_libsWithSyncJniLibsForDebug
:CordovaLib:bundleDebug
:CordovaLib:preReleaseBuild UP-TO-DATE
:prepareOrgApacheCordovaCordovaLib623DebugLibrary
:CordovaLib:checkReleaseManifest
:CordovaLib:prepareReleaseDependencies
:CordovaLib:compileReleaseAidl
:CordovaLib:compileReleaseNdk NO-SOURCE
:CordovaLib:copyReleaseLint NO-SOURCE
:CordovaLib:mergeReleaseShaders
:CordovaLib:compileReleaseShaders
:CordovaLib:generateReleaseAssets
:CordovaLib:mergeReleaseAssets
:CordovaLib:mergeReleaseProguardFiles
:CordovaLib:packageReleaseRenderscript NO-SOURCE
:CordovaLib:compileReleaseRenderscript
:CordovaLib:generateReleaseResValues
:CordovaLib:generateReleaseResources
:CordovaLib:packageReleaseResources
:CordovaLib:processReleaseManifest
:CordovaLib:generateReleaseBuildConfig
:CordovaLib:processReleaseResources
:CordovaLib:generateReleaseSources
:CordovaLib:incrementalReleaseJavaCompilationSafeguard
:CordovaLib:compileReleaseJavaWithJavac
:CordovaLib:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processReleaseJavaRes NO-SOURCE
:CordovaLib:transformResourcesWithMergeJavaResForRelease
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease
:CordovaLib:mergeReleaseJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForRelease
:CordovaLib:transformNative_libsWithSyncJniLibsForRelease
:CordovaLib:bundleRelease
:prepareAndroidArchLifecycleRuntime100Library
:prepareComAndroidSupportAnimatedVectorDrawable2540Library
:prepareComAndroidSupportAppcompatV72540Library
:prepareComAndroidSupportCustomtabs2610Library
:prepareComAndroidSupportSupportCompat2610Library
:prepareComAndroidSupportSupportCoreUi2610Library
:prepareComAndroidSupportSupportCoreUtils2610Library
:prepareComAndroidSupportSupportFragment2610Library
:prepareComAndroidSupportSupportMediaCompat2610Library
:prepareComAndroidSupportSupportV42610Library
:prepareComAndroidSupportSupportVectorDrawable2540Library
:prepareComGoogleAndroidGmsPlayServicesBase1140Library
:prepareComGoogleAndroidGmsPlayServicesBasement1140Library
:prepareComGoogleAndroidGmsPlayServicesGcm1140Library
:prepareComGoogleAndroidGmsPlayServicesIid1140Library
:prepareComGoogleAndroidGmsPlayServicesLocation1140Library
:prepareComGoogleAndroidGmsPlayServicesTasks1140Library
:prepareComOnesignalOneSignal361Library
:prepareDebugDependencies
:compileDebugAidl
:compileDebugRenderscript
:generateDebugResources
:mergeDebugResources
:processDebugManifestD:\Temp\test\platforms\android\AndroidManifest.xml:28:13-35 Error:
        Attribute meta-data#android.support.VERSION@value value=(25.4.0) from [com.android.support:appcompat-v7:25.4.0] AndroidManifest.xml:28:13-35
        is also present at [com.android.support:support-v4:26.1.0] AndroidManifest.xml:28:13-35 value=(26.1.0).
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to override.


See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.

:processDebugManifest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed : Attribute meta-data#android.support.VERSION@value value=(25.4.0) from [com.android.support:appcompat-v7:25.4.0] AndroidManifest.xml:28:13-35
        is also present at [com.android.support:support-v4:26.1.0] AndroidManifest.xml:28:13-35 value=(26.1.0).
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to override.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1 mins 27.625 secs
Error: cmd: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
D:\Temp\test\platforms\android\AndroidManifest.xml:28:13-35 Error:
        Attribute meta-data#android.support.VERSION@value value=(25.4.0) from [com.android.support:appcompat-v7:25.4.0] AndroidManifest.xml:28:13-35
        is also present at [com.android.support:support-v4:26.1.0] AndroidManifest.xml:28:13-35 value=(26.1.0).
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to override.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processDebugManifest'.
> Manifest merger failed : Attribute meta-data#android.support.VERSION@value value=(25.4.0) from [com.android.support:appcompat-v7:25.4.0] AndroidManifest.xml:28:13-35
        is also present at [com.android.support:support-v4:26.1.0] AndroidManifest.xml:28:13-35 value=(26.1.0).
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:26:9-28:38 to override.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

$ cordova plugin add cordova-android-support-gradle-release --variable ANDROID_SUPPORT_VERSION=24.+
Installing "cordova-android-support-gradle-release" for android
ANDROID_HOME=C:\Users\Admin\AppData\Local\Android\sdk
JAVA_HOME=C:\Program Files\Java\jdk1.8.0_101
Subproject Path: CordovaLib
Parallel execution with configuration on demand is an incubating feature.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_6nhbipi8n96npr5m0yzptpel7.run(D:\Temp\test\platforms\android\build.gradle:138)
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
:clean
:CordovaLib:clean

BUILD SUCCESSFUL

Total time: 2.23 secs
Subproject Path: CordovaLib
Adding cordova-android-support-gradle-release to package.json
Saved plugin info for "cordova-android-support-gradle-release" to config.xml
cordova-android-support-gradle-release : WROTE  D:\Temp\test\platforms\android\build.gradle  >  24.+

$ cordova build android
cordova-android-support-gradle-release  ANDROID_SUPPORT_VERSION:  24.+
cordova-android-support-gradle-release : WROTE  D:\Temp\test\platforms\android\build.gradle  >  24.+
ANDROID_HOME=C:\Users\Admin\AppData\Local\Android\sdk
JAVA_HOME=C:\Program Files\Java\jdk1.8.0_101
Subproject Path: CordovaLib
Parallel execution with configuration on demand is an incubating feature.
+-----------------------------------------------------------------
| cordova-android-support-gradle-release: 24.+
+-----------------------------------------------------------------
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_6nhbipi8n96npr5m0yzptpel7.run(D:\Temp\test\platforms\android\build.gradle:139)
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
Download https://maven.google.com/com/android/support/support-v4/24.2.1/support-v4-24.2.1.pom
Download https://maven.google.com/com/android/support/appcompat-v7/24.2.1/appcompat-v7-24.2.1.pom
Download https://maven.google.com/com/android/support/customtabs/24.2.1/customtabs-24.2.1.pom
Download https://maven.google.com/com/android/support/support-compat/24.2.1/support-compat-24.2.1.pom
Download https://maven.google.com/com/android/support/support-media-compat/24.2.1/support-media-compat-24.2.1.pom
Download https://maven.google.com/com/android/support/support-core-utils/24.2.1/support-core-utils-24.2.1.pom
Download https://maven.google.com/com/android/support/support-core-ui/24.2.1/support-core-ui-24.2.1.pom
Download https://maven.google.com/com/android/support/support-fragment/24.2.1/support-fragment-24.2.1.pom
Download https://maven.google.com/com/android/support/support-vector-drawable/24.2.1/support-vector-drawable-24.2.1.pom
Download https://maven.google.com/com/android/support/animated-vector-drawable/24.2.1/animated-vector-drawable-24.2.1.pom
Download https://maven.google.com/com/android/support/support-annotations/24.2.1/support-annotations-24.2.1.pom
:CordovaLib:preBuild
:preBuild
:CordovaLib:preBuild UP-TO-DATE
:preBuild UP-TO-DATE
:CordovaLib:preDebugBuild UP-TO-DATE
:CordovaLib:checkDebugManifest
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:preReleaseBuild UP-TO-DATE
:CordovaLib:prepareDebugDependencies
:CordovaLib:compileDebugAidl
:generateDebugBuildConfig
:CordovaLib:compileDebugNdk NO-SOURCE
:CordovaLib:compileLint
:generateDebugResValues
:CordovaLib:copyDebugLint NO-SOURCE
:incrementalDebugJavaCompilationSafeguard
:CordovaLib:mergeDebugShaders
:CordovaLib:compileDebugShaders
:compileDebugNdk NO-SOURCE
:mergeDebugShaders
:CordovaLib:generateDebugAssets
:CordovaLib:mergeDebugAssets
:compileDebugShaders
:CordovaLib:mergeDebugProguardFiles UP-TO-DATE
:CordovaLib:packageDebugRenderscript
:generateDebugAssets
:processDebugJavaRes
:CordovaLib:packageDebugRenderscript NO-SOURCE
:CordovaLib:compileDebugRenderscript
:processDebugJavaRes NO-SOURCE
:validateSigningDebug
:CordovaLib:generateDebugResValues
:CordovaLib:generateDebugResources
:CordovaLib:packageDebugResources
:CordovaLib:processDebugManifest
:CordovaLib:generateDebugBuildConfig
:CordovaLib:processDebugResources
:CordovaLib:generateDebugSources
:CordovaLib:incrementalDebugJavaCompilationSafeguard
:CordovaLib:compileDebugJavaWithJavac
:CordovaLib:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processDebugJavaRes NO-SOURCE
:CordovaLib:transformResourcesWithMergeJavaResForDebug
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForDebug
:CordovaLib:mergeDebugJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForDebug
:CordovaLib:transformNative_libsWithSyncJniLibsForDebug
:CordovaLib:bundleDebug
:CordovaLib:preReleaseBuild UP-TO-DATE
:prepareOrgApacheCordovaCordovaLib623DebugLibrary
:CordovaLib:checkReleaseManifest
:CordovaLib:prepareReleaseDependencies
:CordovaLib:compileReleaseAidl
:CordovaLib:compileReleaseNdk NO-SOURCE
:CordovaLib:copyReleaseLint NO-SOURCE
:CordovaLib:mergeReleaseShaders
:CordovaLib:compileReleaseShaders
:CordovaLib:generateReleaseAssets
:CordovaLib:mergeReleaseAssets
:CordovaLib:mergeReleaseProguardFiles UP-TO-DATE
:CordovaLib:packageReleaseRenderscript NO-SOURCE
:CordovaLib:compileReleaseRenderscript
:CordovaLib:generateReleaseResValues
:CordovaLib:generateReleaseResources
:CordovaLib:packageReleaseResources
:CordovaLib:processReleaseManifest
:CordovaLib:generateReleaseBuildConfig
:CordovaLib:processReleaseResources
:CordovaLib:generateReleaseSources
:CordovaLib:incrementalReleaseJavaCompilationSafeguard
:CordovaLib:compileReleaseJavaWithJavac
:CordovaLib:compileReleaseJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
:CordovaLib:processReleaseJavaRes NO-SOURCE
:CordovaLib:transformResourcesWithMergeJavaResForRelease
:CordovaLib:transformClassesAndResourcesWithSyncLibJarsForRelease
:CordovaLib:mergeReleaseJniLibFolders
:CordovaLib:transformNative_libsWithMergeJniLibsForRelease
:CordovaLib:transformNative_libsWithSyncJniLibsForRelease
:CordovaLib:bundleRelease
:prepareComAndroidSupportAnimatedVectorDrawable2421Library
:prepareComAndroidSupportAppcompatV72421Library
:prepareComAndroidSupportCustomtabs2421Library
:prepareComAndroidSupportSupportCompat2421Library
:prepareComAndroidSupportSupportCoreUi2421Library
:prepareComAndroidSupportSupportCoreUtils2421Library
:prepareComAndroidSupportSupportFragment2421Library
:prepareComAndroidSupportSupportMediaCompat2421Library
:prepareComAndroidSupportSupportV42421Library
:prepareComAndroidSupportSupportVectorDrawable2421Library
:prepareComGoogleAndroidGmsPlayServicesBase1140Library
:prepareComGoogleAndroidGmsPlayServicesBasement1140Library
:prepareComGoogleAndroidGmsPlayServicesGcm1140Library
:prepareComGoogleAndroidGmsPlayServicesIid1140Library
:prepareComGoogleAndroidGmsPlayServicesLocation1140Library
:prepareComGoogleAndroidGmsPlayServicesTasks1140Library
:prepareComOnesignalOneSignal361Library
:prepareDebugDependencies
:compileDebugAidl
:compileDebugRenderscript
:generateDebugResources
:mergeDebugResources
:processDebugManifest
:processDebugResources
:generateDebugSources
:compileDebugJavaWithJavac
:compileDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\Temp\test\platforms\android\src\cordova\plugins\Diagnostic.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:compileDebugSources
:mergeDebugAssets
:transformClassesWithDexForDebug
:mergeDebugJniLibFolders
:transformNative_libsWithMergeJniLibsForDebug
:transformNative_libsWithStripDebugSymbolForDebug
:transformResourcesWithMergeJavaResForDebug
:packageDebug
:assembleDebug
:cdvBuildDebug

BUILD SUCCESSFUL

Total time: 53.989 secs
Built the following apk(s):
        D:/Temp/test/platforms/android/build/outputs/apk/android-debug.apk

```
</details>

<br/>

> 2. Could you explain a little on why the latest version clashes with plugins?

The Gradle versioning of the Support Library is a moving target: it depends which version other plugins specify as to whether you'll encounter a build issue, so my policy with this plugin has been to specify the most recent version at the time of publishing. Hence v3.1.0 (from June 2016) [specified v23](https://github.com/dpa99c/cordova-diagnostic-plugin/blob/3.1.0/plugin.xml#L56) whereas v3.6.7 [specifies v25](https://github.com/dpa99c/cordova-diagnostic-plugin/blob/3.6.7/plugin.xml#L101). The actual [most recent release](https://developer.android.com/topic/libraries/support-library/revisions.html) is v26.

No matter what version I pin in this plugin, it will collide with another plugin that specifies a different version, hence the reason I created `cordova-android-support-gradle-release` which enables you to override the versions specified by installed plugins and align the versions, so the build will succeed. `cordova-android-support-gradle-release` is kept up-to-date to default to the most recent release (currently v26) but you can specify an earlier version using the `ANDROID_SUPPORT_VERSION` variable.

**Comment by ilhamsuaib** [comment]
did u use Android Studio Beta Version?
i have same problem when i use AS beta version, and solved when im using stable version. just change buildToolsVersion and support version from 26.0.1 -> 26.1.0
